wicket-core/src/main/java/org/apache/wicket/resource/FileSystemResource.java

```diff
@@ -0,0 +1,145 @@
+ * 
+ * @author Tobias Soloschenko
+ *
+ */
+public class FileSystemResource extends AbstractResource
+{
+	private static final long serialVersionUID = 1L;
+
+	private final Path path;
+
+	/**
+	 * Creates a new file system resource based on the given path
+	 * 
+	 * @param path
+	 *            the path to be read for the resource
+	 */
+	public FileSystemResource(Path path)
+	{
+		Args.notNull(path, "path");
+		this.path = path;
+	}
+
+	/**
+	 * Creates a new resource response and reads the given path
+	 */
+	@Override
+	protected ResourceResponse newResourceResponse(Attributes attributes)
+	{
+		try
+		{
+			long size = getSize();
+			ResourceResponse resourceResponse = new ResourceResponse();
+			resourceResponse.setContentType(getMimeType());
+			resourceResponse.setAcceptRange(ContentRangeType.BYTES);
+			resourceResponse.setContentLength(size);
+			resourceResponse = configureResourceResponse(resourceResponse);
+			RequestCycle cycle = RequestCycle.get();
+			Long startbyte = cycle.getMetaData(CONTENT_RANGE_STARTBYTE);
+			Long endbyte = cycle.getMetaData(CONTENT_RANGE_ENDBYTE);
+			resourceResponse.setWriteCallback(new PartWriterCallback(getInputStream(), size,
+				startbyte, endbyte));
+			return resourceResponse;
+		}
+		catch (IOException e)
+		{
+			throw new WicketRuntimeException(
+				"An error occured while processing the media resource response", e);
+		}
+	}
+
+	/**
+	 * Configures the resource response by overriding it with additional information
+	 * 
+	 * @param resourceResponse
+	 *            the resource response to be configured
+	 * 
+	 * @return the resource response to be configured
+	 */
+	protected ResourceResponse configureResourceResponse(ResourceResponse resourceResponse)
+	{
+		return resourceResponse;
+	}
+
+	/**
+	 * Gets the size of the resource
+	 * 
+	 * @return the size of the resource
+	 * @throws IOException
+	 *             if the size attribute can't be read
+	 */
+	protected long getSize() throws IOException
+	{
+		return Files.readAttributes(path, BasicFileAttributes.class).size();
+	}
+
+	/**
+	 * Gets the mime type to be used for the response it first uses the URL connection to get the
+	 * mime type and after this the FileTypeDetector SPI is used.
+	 * 
+	 * @return the mime type to be used for the response
+	 * @throws IOException
+	 *             if the mime type could'nt be resoulved
+	 */
+	protected String getMimeType() throws IOException
+	{
+		String mimeType = null;
+		if (Application.exists())
+		{
+			mimeType = Application.get().getMimeType(path.getFileName().toString());
+		}
+		if (mimeType == null)
+		{
+
+			mimeType = Files.probeContentType(path);
+		}
+		return mimeType;
+	}
+
+	/**
+	 * Gets the input stream of the given path
+	 * 
+	 * @return the input stream of the given path
+	 * @throws IOException
+	 *             if there is an exception while receiving the input stream
+	 */
+	protected InputStream getInputStream() throws IOException
```

Comment:
I made all methods which uses Files overridable, so that it can be customized if required.